example/mixed-arch/main.tf

```diff
@@ -2,22 +2,22 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "=2.51.0"
+      version = ">=2.51.0"
     }
     random = {
       source  = "hashicorp/random"
-      version = "=2.3.0"
+      version = ">=2.3.0"
```

Comment:
Same as last comment, replace operator with `~>` and drop last bit of version.

Also note that v3 of this provider has been out for a while, I forget why they had to increment the major version but I've been using v3 with no issues.